src/main/java/core/basesyntax/operations/SupplyOperation.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    private final Storage storage;
+
+    public SupplyOperation(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (storage == null) {
+            throw new IllegalArgumentException("Storage cannot be null.");
+        }
+        if (transaction == null) {
+            throw new IllegalArgumentException("Transaction cannot be null.");
+        }
+        if (transaction.getQuantity() <= 0) {
+            throw new IllegalArgumentException("Quantity must be greater than zero.");
+        }

```

Comment:
It's good that you're validating the `transaction` quantity. However, consider moving this validation to a separate method or class that is responsible for validating `FruitTransaction` objects to keep the `apply` method focused on its primary responsibility and adhere to the Single Responsibility Principle.